Description: Use python3 in tests
 Some upstream tests call the python interpreter in a subprocess, e.g. to test
 the custom `process` module or to run simple in-toto step or inspection
 commands. In debian, where we only build for python3, "python" is not
 available during build and thus makes those tests fail. This patch replaces
 "python" with "python3" where applicable.
Forwarded: not-needed
Origin: upstream, https://github.com/in-toto/in-toto/commit/a7f4baf8d0186fa4a6314264707f4d5d1b3cecdc

Index: in-toto-1.2.0/tests/demo_files/demo.layout.template
===================================================================
--- in-toto-1.2.0.orig/tests/demo_files/demo.layout.template
+++ in-toto-1.2.0/tests/demo_files/demo.layout.template
@@ -25,7 +25,7 @@
             ],
             "name": "untar",
             "run": [
-                "python",
+                "python3",
                 "./scripts/tar",
                 "xfz",
                 "foo.tar.gz"
Index: in-toto-1.2.0/tests/test_in_toto_mock.py
===================================================================
--- in-toto-1.2.0.orig/tests/test_in_toto_mock.py
+++ in-toto-1.2.0/tests/test_in_toto_mock.py
@@ -73,7 +73,7 @@ class TestInTotoMockTool(CliTestCase, Tm
   def test_main_required_args(self):
     """Test CLI command with required arguments. """
 
-    args = ["--name", self.test_step, "--", "python", "--version"]
+    args = ["--name", self.test_step, "--", "python3", "--version"]
     self.assert_cli_sys_exit(args, 0)
 
     self.assertTrue(os.path.exists(self.test_link))
Index: in-toto-1.2.0/tests/test_in_toto_run.py
===================================================================
--- in-toto-1.2.0.orig/tests/test_in_toto_run.py
+++ in-toto-1.2.0/tests/test_in_toto_run.py
@@ -76,7 +76,7 @@ class TestInTotoRunTool(CliTestCase, Tmp
     """Test CLI command with required arguments. """
 
     args = ["--step-name", self.test_step, "--key", self.rsa_key_path, "--",
-        "python", "--version"]
+        "python3", "--version"]
 
     self.assert_cli_sys_exit(args, 0)
     self.assertTrue(os.path.exists(self.test_link_rsa))
@@ -88,7 +88,7 @@ class TestInTotoRunTool(CliTestCase, Tmp
     named_args = ["--step-name", self.test_step, "--key",
         self.rsa_key_path, "--materials", self.test_artifact, "--products",
         self.test_artifact, "--record-streams"]
-    positional_args = ["--", "python", "--version"]
+    positional_args = ["--", "python3", "--version"]
 
     # Test and assert recorded artifacts
     args1 = named_args + positional_args
@@ -134,7 +134,7 @@ class TestInTotoRunTool(CliTestCase, Tmp
     """Test CLI command with metadata directory. """
     tmp_dir = os.path.realpath(tempfile.mkdtemp(dir=os.getcwd()))
     args = ["--step-name", self.test_step, "--key", self.rsa_key_path,
-        "--metadata-directory", tmp_dir, "--", "python", "--version"]
+        "--metadata-directory", tmp_dir, "--", "python3", "--version"]
 
     self.assert_cli_sys_exit(args, 0)
 
@@ -166,7 +166,7 @@ class TestInTotoRunTool(CliTestCase, Tmp
           "-n", self.test_step,
           "--key", key_path,
           "--key-type", key_type]
-      cmd = ["--", "python", "--version"]
+      cmd = ["--", "python3", "--version"]
 
       # Make sure the link file to be generated doesn't already exist
       self.assertFalse(os.path.exists(link_path))
@@ -189,7 +189,7 @@ class TestInTotoRunTool(CliTestCase, Tmp
     """Test CLI command with specified gpg key. """
     args = ["-n", self.test_step,
             "--gpg", self.gpg_key_85da58,
-            "--gpg-home", self.gnupg_home, "--", "python", "--version"]
+            "--gpg-home", self.gnupg_home, "--", "python3", "--version"]
 
     self.assert_cli_sys_exit(args, 0)
     link_filename = FILENAME_FORMAT.format(step_name=self.test_step,
@@ -201,7 +201,7 @@ class TestInTotoRunTool(CliTestCase, Tmp
   def test_main_with_default_gpg_key(self):
     """Test CLI command with default gpg key. """
     args = ["-n", self.test_step,
-            "--gpg", "--gpg-home", self.gnupg_home, "--", "python", "--version"]
+            "--gpg", "--gpg-home", self.gnupg_home, "--", "python3", "--version"]
 
     self.assert_cli_sys_exit(args, 0)
 
Index: in-toto-1.2.0/tests/test_runlib.py
===================================================================
--- in-toto-1.2.0.orig/tests/test_runlib.py
+++ in-toto-1.2.0/tests/test_runlib.py
@@ -517,17 +517,17 @@ class TestInTotoRun(unittest.TestCase, T
   def test_in_toto_run_verify_signature(self):
     """Successfully run, verify signed metadata. """
     link = in_toto_run(self.step_name, None, None,
-        ["python", "--version"], True, self.key)
+        ["python3", "--version"], True, self.key)
     link.verify_signature(self.key)
 
   def test_in_toto_run_no_signature(self):
     """Successfully run, verify empty signature field. """
-    link = in_toto_run(self.step_name, None, None, ["python", "--version"])
+    link = in_toto_run(self.step_name, None, None, ["python3", "--version"])
     self.assertFalse(len(link.signatures))
 
   def test_in_toto_run_with_byproduct(self):
     """Successfully run, verify recorded byproduct. """
-    link = in_toto_run(self.step_name, None, None, ["python", "--version"],
+    link = in_toto_run(self.step_name, None, None, ["python3", "--version"],
         record_streams=True)
 
     # this or clause may seem weird, but given that python 2 prints its version
@@ -540,14 +540,14 @@ class TestInTotoRun(unittest.TestCase, T
 
   def test_in_toto_run_without_byproduct(self):
     """Successfully run, verify byproduct is not recorded. """
-    link = in_toto_run(self.step_name, None, None, ["python", "--version"],
+    link = in_toto_run(self.step_name, None, None, ["python3", "--version"],
         record_streams=False)
     self.assertFalse(len(link.signed.byproducts.get("stdout")))
 
   def test_in_toto_run_compare_dumped_with_returned_link(self):
     """Successfully run, compare dumped link is equal to returned link. """
     link = in_toto_run(self.step_name, [self.test_artifact],
-        [self.test_artifact], ["python", "--version"], True, self.key)
+        [self.test_artifact], ["python3", "--version"], True, self.key)
     link_dump = Metablock.load(
         FILENAME_FORMAT.format(step_name=self.step_name, keyid=self.key["keyid"]))
     self.assertEqual(repr(link), repr(link_dump))
@@ -557,7 +557,7 @@ class TestInTotoRun(unittest.TestCase, T
      compare dumped link is equal to returned link"""
     tmp_dir = os.path.realpath(tempfile.mkdtemp(dir=os.getcwd()))
     link = in_toto_run(self.step_name, [self.test_artifact],
-        [self.test_artifact], ["python", "--version"], True, self.key,
+        [self.test_artifact], ["python3", "--version"], True, self.key,
         metadata_directory=tmp_dir)
     file_path = os.path.join(tmp_dir, FILENAME_FORMAT.format(step_name=self.step_name,
         keyid=self.key["keyid"]))
@@ -569,13 +569,13 @@ class TestInTotoRun(unittest.TestCase, T
      compare the signed is equal"""
     tmp_dir = os.path.realpath(tempfile.mkdtemp(dir=os.getcwd()))
     in_toto_run(self.step_name, [self.test_artifact], [self.test_artifact],
-        ["python", "--version"], True, self.key, metadata_directory=tmp_dir)
+        ["python3", "--version"], True, self.key, metadata_directory=tmp_dir)
     file_path = os.path.join(tmp_dir, FILENAME_FORMAT.format(step_name=self.step_name,
         keyid=self.key["keyid"]))
     link_dump_with_md = Metablock.load(file_path)
 
     in_toto_run(self.step_name, [self.test_artifact], [self.test_artifact],
-        ["python", "--version"], True, self.key)
+        ["python3", "--version"], True, self.key)
     link_dump_without_md = Metablock.load(
         FILENAME_FORMAT.format(step_name=self.step_name, keyid=self.key["keyid"]))
     self.assertEqual(repr(link_dump_with_md.signed),
@@ -584,13 +584,13 @@ class TestInTotoRun(unittest.TestCase, T
   def test_in_toto_run_verify_recorded_artifacts(self):
     """Successfully run, verify properly recorded artifacts. """
     link = in_toto_run(self.step_name, [self.test_artifact],
-        [self.test_artifact], ["python", "--version"])
+        [self.test_artifact], ["python3", "--version"])
     self.assertEqual(list(link.signed.materials.keys()),
         list(link.signed.products.keys()), [self.test_artifact])
 
   def test_in_toto_run_verify_workdir(self):
     """Successfully run, verify cwd. """
-    link = in_toto_run(self.step_name, [], [], ["python", "--version"],
+    link = in_toto_run(self.step_name, [], [], ["python3", "--version"],
         record_environment=True)
     self.assertEqual(link.signed.environment["workdir"],
         os.getcwd().replace("\\", "/"))
@@ -608,7 +608,7 @@ class TestInTotoRun(unittest.TestCase, T
 
       # Call in_toto_run and record artifacts as materials and products
       # with line ending normalization on
-      link = in_toto_run(self.step_name, paths, paths, ["python", "--version"],
+      link = in_toto_run(self.step_name, paths, paths, ["python3", "--version"],
           normalize_line_endings=True).signed
 
       # Check that all three hashes in materials and products are equal
@@ -626,18 +626,18 @@ class TestInTotoRun(unittest.TestCase, T
     """Fail run, passed key is not properly formatted. """
     with self.assertRaises(securesystemslib.exceptions.FormatError):
       in_toto_run(self.step_name, None, None,
-          ["python", "--version"], True, "this-is-not-a-key")
+          ["python3", "--version"], True, "this-is-not-a-key")
 
   def test_in_toto_wrong_key(self):
     """Fail run, passed key is a public key. """
     with self.assertRaises(securesystemslib.exceptions.FormatError):
       in_toto_run(self.step_name, None, None,
-          ["python", "--version"], True, self.key_pub)
+          ["python3", "--version"], True, self.key_pub)
 
   def test_nonexistent_directory(self):
     """Fail run, passed metadata_directory not exist. """
     with self.assertRaises(FileNotFoundError):
-      in_toto_run(self.step_name, None, None, ["python", "--version"],
+      in_toto_run(self.step_name, None, None, ["python3", "--version"],
           True, self.key, metadata_directory='nonexistentDir')
 
   def test_not_a_directory(self):
@@ -647,7 +647,7 @@ class TestInTotoRun(unittest.TestCase, T
     os.close(fd)
     # Windows will raise FileNotFoundError instead of NotADirectoryError
     with self.assertRaises((NotADirectoryError, FileNotFoundError)):
-      in_toto_run(self.step_name, None, None, ["python", "--version"],
+      in_toto_run(self.step_name, None, None, ["python3", "--version"],
           True, self.key, metadata_directory=path)
     os.remove(path)
 
@@ -658,7 +658,7 @@ class TestInTotoRun(unittest.TestCase, T
     # make the directory read only
     os.chmod(tmp_dir, stat.S_IREAD)
     with self.assertRaises(PermissionError):
-      in_toto_run(self.step_name, None, None, ["python", "--version"],
+      in_toto_run(self.step_name, None, None, ["python3", "--version"],
           True, self.key, metadata_directory=tmp_dir)
     os.rmdir(tmp_dir)
 
Index: in-toto-1.2.0/tests/test_verifylib.py
===================================================================
--- in-toto-1.2.0.orig/tests/test_verifylib.py
+++ in-toto-1.2.0/tests/test_verifylib.py
@@ -106,7 +106,7 @@ class TestRunAllInspections(unittest.Tes
         "steps": [],
         "inspect": [{
           "name": "touch-bar",
-          "run": ["python", os.path.join(scripts_directory, "touch"), "bar"],
+          "run": ["python3", os.path.join(scripts_directory, "touch"), "bar"],
         }]
       })
 
@@ -142,7 +142,7 @@ class TestRunAllInspections(unittest.Tes
         "steps": [],
         "inspect": [{
           "name": "non-zero-inspection",
-          "run": ["python", "./scripts/expr", "1", "/", "0"],
+          "run": ["python3", "./scripts/expr", "1", "/", "0"],
         }]
     })
     with self.assertRaises(BadReturnValueError):
@@ -854,7 +854,7 @@ class TestInTotoVerify(unittest.TestCase
 
     # dump layout with failing inspection retval
     layout = copy.deepcopy(layout_template)
-    layout.signed.inspect[0].run = ["python", "./scripts/expr", "1", "/", "0"]
+    layout.signed.inspect[0].run = ["python3", "./scripts/expr", "1", "/", "0"]
     layout.sign(alice)
     layout.dump(self.layout_failing_inspection_retval)
 
